javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/RemappingUrlConnection.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling;
+
+import static io.opentelemetry.javaagent.tooling.ShadingRemapper.rule;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.security.Permission;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import net.bytebuddy.jar.asm.ClassReader;
+import net.bytebuddy.jar.asm.ClassWriter;
+import net.bytebuddy.jar.asm.commons.ClassRemapper;
+
+public class RemappingUrlConnection extends URLConnection {
+  // We need to prefix the names to prevent the gradle shadowJar relocation rules from touching
+  // them. It's possible to do this by excluding this class from shading, but it may cause issue
+  // with transitive dependencies down the line.
+  private static final ShadingRemapper remapper =
+      new ShadingRemapper(
+          rule("#io.opentelemetry.api", "#io.opentelemetry.javaagent.shaded.io.opentelemetry.api"),
+          rule(
+              "#io.opentelemetry.context",
+              "#io.opentelemetry.javaagent.shaded.io.opentelemetry.context"),
+          rule(
+              "#io.opentelemetry.extension.aws",
+              "#io.opentelemetry.javaagent.shaded.io.opentelemetry.extension.aws"),
```

Comment:
Should we add `io.opentelemetry.semconv` and `io.opentelemetry.instrumentation` too?

Reply:
Indeed, will do